### PR TITLE
Fix model download failure for backend tests

### PR DESCRIPTION
**Pull request by gongsu832**
- add targets to pre-download ONNX model files for backend tests
- add -l|--list option to test.py to list model/node tests used for a backend test configuration
- reimplement strtobool since distutils is deprecated in python 3.10 and will be removed in 3.12

Signed-off-by: Gong Su <gong_su@hotmail.com>

### CODE REVIEW

File: test/backend/CMakeLists.txt [hunk 1]
```diff
@@ -83,21 +83,32 @@ else()
...
+include(setup_model_download.cmake)
+
 # ${ONNX_HOME} is the directory where onnx downloads real model files.
 # Model files are saved under ${ONNX_HOME}/models/model_name/model.onnx.
 # C/C++ and JNI tests run in parallel so they must use a different
 # ONNX_HOME to avoid conflicts.
+#
+# We now have our own model download targets and different backend tests
+# share the pre-downloaded model files. So we no longer need a different
+# ONNX_HOME for C/C++ and JNI tests.
```
Comment (tungld) [inline test/backend/CMakeLists.txt:95]: This is cool. Thanks!

**Review by tungld — APPROVED** [review]
LGTM.

Should we recommend developers to run `make check-onnx-backend-numerical` to test everything before submitting a PR?

File: test/backend/CMakeLists.txt (hunk 1 above)
Comment (AlexandreEichenberger) [reply test/backend/CMakeLists.txt:95]: Thanks, so looking forward to have faster more reliable CIs. Big thanks @gongsu832

**Comment by gongsu832** [comment]
`check-onnx-backend` and `check-onnx-numerical` have been merged into a single `check-onnx-backend-numerical` to have proper parallel behavior and dependency resolution. See comments in `test/CMakeLists.txt` for details. Individual `check-onnx-backend` and `check-onnx-numerical` still exist.

**Comment by gongsu832** [comment]
> Should we recommend developers to run `make check-onnx-backend-numerical` to test everything before submitting a PR?

Individual `check-onnx-backend` and `check-onnx-numerical` still exist so it doesn't really matter that much how people want to run their tests locally.

**Pull request merged**